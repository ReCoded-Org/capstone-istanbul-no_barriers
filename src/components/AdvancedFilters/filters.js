// filterParameters should be fetched from database
const filters = [
  {
    filterName: "resources.AdvancedFilters.city",
    filterParameters: ["ankara", "istanbul"],
  },
  {
    filterName: "resources.AdvancedFilters.ngoName",
    filterParameters: ["teng", "abc"],
  },
  {
    filterName: "resources.AdvancedFilters.service",
    filterParameters: ["healthcare", "education"],
  },
  {
    filterName: "resources.AdvancedFilters.rating",
    filterParameters: [1, 2, 3, 4, 5],
  },
];

export default filters;
