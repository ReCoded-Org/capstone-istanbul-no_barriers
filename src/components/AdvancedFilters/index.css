/* Color Variables */
:root {
  --main-filter-bg-color: #00374d;
  --main-filter-text-color: #d7f7fa;
  --orange-link: #f2c94c;
  --info-bg-color: #29ccd8;
  --dark-orange: #d88315;
  --orange: #fc9e01;
}

.advanceFilters {
  background-color: var(--main-filter-bg-color);
  color: var(--main-filter-text-color);
  padding: 2rem;
}

.advanceFiltersTitle {
  border-bottom: 1px solid var(--main-filter-text-color);
  padding-left: 1rem;
  margin-bottom: 1rem;
}

.moreInfoFilters {
  border-radius: 50%;
  background-color: var(--info-bg-color);
  padding: 0.5rem 1rem;
  margin: -0.4rem 1rem 1rem;
}

.moreInfoFilters:hover {
  background-color: var(--orange-link);
}

.filter {
  margin: none;
  font-weight: bold;
  padding: 0.5rem 0 0;
}

.filter:hover {
  background-color: var(--orange-link);
  color: var(--main-filter-bg-color);
}

.circle {
  position: relative;
  background-color: var(--main-filter-text-color);
  color: var(--main-filter-bg-color);
  border-radius: 50%;
  width: 1.5rem;
  height: 1.5rem;
  margin: 0 0.5rem;
}

.filterPlusSign {
  position: absolute;
  top: -0.1rem;
  left: 0.4rem;
}

.filterButtons {
  margin-top: 1.5rem;
}

.clearAll {
  margin-right: 2rem;
}

.clearAll:hover {
  color: var(--orange-link);
}

.applyFilters,
.applyFilters:hover,
.applyFilters:focus {
  background-color: var(--orange);
  border: none;
}

/* filter all button active state style */
.applyFilters.btn-primary:not(:disabled):not(.disabled):active {
  background-color: var(--dark-orange);
}
