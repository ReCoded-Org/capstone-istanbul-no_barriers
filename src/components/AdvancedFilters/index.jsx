import React from 'react';

const AdvancedFilters = () => {
    return (
        <div>
            
        </div>
    );
}

export default AdvancedFilters;
