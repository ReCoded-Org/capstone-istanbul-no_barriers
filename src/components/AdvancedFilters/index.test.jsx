import React from "react";
import renderer from "react-test-renderer";
import AdvancedFilters from "./index";

const ngos = [
  { city: "Ankara", service: "education", name: "ABC", rating: "1" },
];
const filteredNgos = [];
const setFilteredNgos = (ngos) => filteredNgos.push(ngos);
const filterOptions = {};

// Advance Filters component's functionalities - START
const applyAdvancedFilters = (filterOptions, ngos, setFilteredNgos) => {
  setFilteredNgos([]);
  for (let i = 0; i < ngos.length; i++) {
    const isAllOptionsIncluded =
      filterOptions.cities.includes(ngos[i].city) &&
      filterOptions.services.includes(ngos[i].service) &&
      filterOptions.names.includes(ngos[i].name) &&
      filterOptions.ratings.includes(ngos[i].rating);

    if (isAllOptionsIncluded) {
      setFilteredNgos((prevState) => [...prevState, ngos[i]]);
    }
  }
};

it("AdvancedFilters renders correctly", () => {
  const tree = renderer
    .create(
      <AdvancedFilters
        applyAdvancedFilters={() =>
          applyAdvancedFilters(filterOptions, ngos, setFilteredNgos)
        }
        filterOptions={filterOptions}
        setFilteredNgos={setFilteredNgos}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
