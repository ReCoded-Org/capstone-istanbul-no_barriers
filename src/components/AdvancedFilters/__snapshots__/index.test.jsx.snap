// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedFilters renders correctly 1`] = `
<div
  className="advancedFilters"
>
  <div
    className="container"
  >
    <div
      className="advancedFiltersTitleSection d-flex justify-content-between align-items-baseline pb-2"
    >
      <h5
        className="advancedFiltersTitle"
      >
        resources.AdvancedFilters.title
      </h5>
      <span
        className="moreInfoFilters"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        ?
      </span>
    </div>
    <div
      className="filterButtons d-flex justify-content-center align-items-baseline"
    >
      <p
        className="clearAll"
        onClick={[Function]}
      >
        resources.AdvancedFilters.clearAll
      </p>
      <button
        className="applyFilters shadow-none btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        resources.AdvancedFilters.applyFilters
      </button>
    </div>
  </div>
</div>
`;
